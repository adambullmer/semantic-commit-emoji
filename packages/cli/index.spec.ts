/// <reference types="@types/jest" />
import index from "./index";
import { Config } from "semantic-commit-emoji-config";

const config: Config = {
  withSpace: false,
  automaticTypes: {
    fixup: "",
    revert: "rewind",
    merge: "twisted_rightwards_arrows",
    version: "bookmark",
  },
  conventionalTypes: {
    feat: "sparkles",
    fix: "bug",
    chore: "wrench",
  },
};

describe("Semantic Commit Emoji Processor", () => {
  it.each([
    ["a normal commit message"],
    ["fix: A commit with a semantic type"],
    [":wrench:chore: A commit with an existing emoji"],
    ["feat!: A commit with a breaking change notation"],
    [
      `feat: A multiline commit

A summary about the commit in detail

Resolves #2`,
    ],
    ["v1.2.3"],
    ["Merge pull request #1"],
    ["Revert: Merge pull request #1"],
    ["fixup! :bug:fix: A commit with a semantic type"],
    ["build: An unsupported commit type"],
  ])('"%s")', (input) => {
    expect(index(config, input)).toMatchSnapshot();
  });
});
