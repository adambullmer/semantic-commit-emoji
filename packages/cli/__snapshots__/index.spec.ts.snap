// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Semantic Commit Emoji Processor > ":wrench:chore: A commit with an existing emoji") 1`] = `":wrench:chore: A commit with an existing emoji"`;

exports[`Semantic Commit Emoji Processor > "Merge pull request #1") 1`] = `":twisted_rightwards_arrows:Merge pull request #1"`;

exports[`Semantic Commit Emoji Processor > "Revert: Merge pull request #1") 1`] = `":rewind:Revert: Merge pull request #1"`;

exports[`Semantic Commit Emoji Processor > "a normal commit message") 1`] = `"a normal commit message"`;

exports[`Semantic Commit Emoji Processor > "build: An unsupported commit type") 1`] = `"build: An unsupported commit type"`;

exports[`Semantic Commit Emoji Processor > "feat!: A commit with a breaking change notation") 1`] = `":sparkles:feat!: A commit with a breaking change notation"`;

exports[`Semantic Commit Emoji Processor > "feat: A multiline commit

A summary about the commit in detail

Resolves #2") 1`] = `
":sparkles:feat: A multiline commit

A summary about the commit in detail

Resolves #2"
`;

exports[`Semantic Commit Emoji Processor > "fix: A commit with a semantic type") 1`] = `":bug:fix: A commit with a semantic type"`;

exports[`Semantic Commit Emoji Processor > "fixup! :bug:fix: A commit with a semantic type") 1`] = `"fixup! :bug:fix: A commit with a semantic type"`;

exports[`Semantic Commit Emoji Processor > "v1.2.3") 1`] = `":bookmark:v1.2.3"`;
